### ISSUE TITLE
[Bug]:  subscription failure OR just autologout = it's necessary to configure everything AGAIN

### ISSUE THREAD

**Issue by intari** [httptoolkit/httptoolkit#604]
### Has this been reported before?

- [X] I have checked for existing reports of this issue

### Repro steps

Open desktop app, subscribe, configure custom CAs, TLS passtroughs, custom mocks, make next subscription fail = everything is gone and need to be resetup again. 
Also, Windows app sometimes just logs out and after relogin - everything is gone again. 
If app wants to reset configuration - it should be at least backupable, if Pro-level configuration is no longer available because of paddle issues - it should just be blocked and not deleted from configuration.

### How often does this bug happen?

Every time

### The desktop OS you're using

Windows 11 Pro 23H2

**Comment by pimterry** [comment]
Yes, indeed. This is a duplicate of #240, which you've commented on in the past.

This is a bit harder to do reliably that it sounds, though any contributions to help fix this are welcome! Note that Pro is totally free for contributors, so you'd avoid this problem entirely anyway.

In general, the 'right' solution is to not use HTTP Toolkit without a Pro subscription if you're trying to use Pro features. The Pro subscriptions do send plenty of emails if payments fail or similar, and nothing will reset if you don't try to use the app in that time.

> Also, Windows app sometimes just logs out and after relogin - everything is gone again.

That's more concerning though, and definitely shouldn't happen. I'm not aware of issues that would cause the windows app to lose the login state, but that definitely sounds like a bug. Can you share any more info about that? How frequently does this happen? Is there anything that triggers it? Does it happen even if you're using the app frequently or only after a long time without opening it or similar?

To be honest that's a bigger issue, since that login state lives in the same store as all the rules etc. Even if we had a good way to persist the rules for users without Pro accounts, if all the app's state is being reset somehow then that would be lost anyway.

As a short-term fix, a contribution to add import/export button to the settings page so you can back that up (and share it with others) would be welcome, that makes perfect sense to me. As a hacky workaround in the meantime, you can always access and modify all of HTTP Toolkit's data through the dev tools (View -> Toggle developer tools) within 'Local Storage' in the 'Application' tab. You can manually back up and restore data from there if required (this isn't officially supported, but I'd generally expect it to work pretty reliably).
